app/services/platforms/youtube.ts

```diff
@@ -155,8 +188,32 @@ export class YoutubeService extends Service implements IPlatformService {
       .then(handleErrors)
       .then(response => response.json())
       .then(json => {
-        const youtubeDomain = mode === 'day' ? 'https://youtube.com' : 'https://gaming.youtube.com';
+        const youtubeDomain =
+          mode === 'day' ? 'https://youtube.com' : 'https://gaming.youtube.com';
         return `${youtubeDomain}/live_chat?v=${json.items[0].id}&is_popout=1`;
       });
   }
 }
+
+/**
+ * You can use this decorator to ensure that a
+ * fresh youtube api token is available
+ */
+function requiresToken() {
+  return (target: any, methodName: string, descriptor: PropertyDescriptor) => {
+    const original = descriptor.value;
+    return {
+      ...descriptor,
+      value(...args: any[]) {
+        return original.apply(target.constructor.instance, args)
+        .catch((error:Response) => {
+          if (error.status === 401) {
+            return target.fetchNewToken().then(() => {
+              return original.apply(target.constructor.instance, args);
+            });
+          }
```

Comment:
Let's add an `else` case here that returns a rejected promise with the response object.  Otherwise this decorator is just going to eat any non-401 errors and not surface them.

Reply:
fixed in dc4ea8348996ab54bd0d1c2307f2c6abd69100dc